### ISSUE TITLE
Gyb checksum formula

### ISSUE THREAD

**Issue by jpcima** [Wohlstand/OPN2BankEditor#61]
By chance I found the checksum formula of this file format which was previouly unknown.
The project vgmpre has it, where it is known as "nineko's crazy checksum formula".

Judging from code, I guess we can say it's a somewhat fitting description.
https://github.com/vgmtool/vgm2pre/blob/f4f917df35d531512292541234a5c1722b8af96f/vgm2pre/fmt/gyb.hpp#L30-L70

**Comment by Wohlstand** [comment]
So, let's try to implement this and check will midi2vgm accept the result?

**Comment by jpcima** [comment]
Oh it's a minor thing, I wanted to keep note because I've found it during research of formats.
Are other software known to use this format?

**Comment by Wohlstand** [comment]
>  Are other software known to use this format?

Idk, mainly those midi2vgm and it's editor

**Comment by jpcima** [comment]
I found again an old post where ValleyBell talks of a newer revision of Gyb known as v3 (Nov 2018).
It is suggested to have support of bank MSB+LSB through linked editor software.
This software has source code with the Gyb loader. At first look, it has a case for `FileVer` based on 3 different versions.
http://forums.sonicretro.org/index.php?s=9007c6bcd222e60acc7e3e2f044313ce&showtopic=25337&view=findpost&p=934762

**Comment by ValleyBell** [comment]
If you want a GYB checksum implementation, feel free to use the one from *2612_checksum.c* that's included in the [mid2smps 0.3.5 source code](http://vgmrips.net/programs/non-vgm/mid2smps_0350_src.7z) (subfolder OPNInsEdit)

So far the GYB format is only used by midi2vgm YM2612 and mid2smps. The latter is pretty popular in the Sonic hacking scene.

**Comment by Wohlstand** [comment]
> So far the GYB format is only used by midi2vgm YM2612 and mid2smps. The latter is pretty popular in the Sonic hacking scene.

I see, BTW, some recent time I made my own MIDI2VGM with using of libOPNMIDI core:
- The tool: https://github.com/Wohlstand/libOPNMIDI/blob/master/utils/midi2vgm/midi2vgm.cpp
- The actual writer: https://github.com/Wohlstand/libOPNMIDI/blob/master/src/chips/vgm_file_dumper.cpp
Processing of channels and music playing libOPNMIDI is doing. Yeah, libOPNMIDI initially is the MIDI synthesizer over OPN2 and OPNA chips, however, knowing some tricks I made the MIDI2VGM tool with basing on capturing of OPN commands and sending them into VGM file. Files are appearing non-optimized and their size would have more than 1 MB. However, your `vgm_cmp` squishes it into a few kilobytes size (except files are getting too many arpeggios due to voice channels overuse because of notes mess).
